graphstorm-processing/graphstorm_processing/graph_loaders/row_count_utils.py

```diff
@@ -229,10 +234,15 @@ def _add_counts_for_features(self, top_level_key: str, edge_or_node_type_key: st
                     self.metadata_dict[top_level_key].keys(),
                 )
                 continue
-            for feature_name, feature_data_dict in self.metadata_dict[top_level_key][
-                type_name
-            ].items():
-                relative_file_list = feature_data_dict["data"]  # type: Sequence[str]
+
+            # Create a new dictionary for this type's features
+            type_features = self.metadata_dict[top_level_key][type_name]
+
+            for feature_name, feature_data in type_features.items():
+                # Create a copy of the feature data to avoid modifying shared references
+                feature_data_dict = feature_data.copy()
```

Comment:
Still metadata here?

Reply:
Yes, feature_data refers to to the `data` entry in a feature, a dict with two or three entries, one string for `format`, one list of str for `files` and optionally one string for separator.